ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/ASTConverter.java

```diff
@@ -621,6 +644,109 @@ private ASTNode pkFkHint(int fkTableIndex, boolean nonFkSideIsFiltered) {
     }
   }
 
+  private ASTNode createASTLateralView(TableFunctionScan tfs, Schema s,
+      QueryBlockInfo tableFunctionSource, String sqAlias) {
+    // In the case where the RelNode is a HiveTableFunctionScan, first we check
+    // to see if we can't optimize out the lateral view operator. We can optimize the
+    // operator out if only the udtf fields are grabbed out of the RelNode.  If any
+    // of the base table fields need to be grabbed out, then a 'join' needs to be done
+    // and we need the lateral view.
+    //
```

Comment:
The comment seems misplaced here. Maybe it is also a bit redundant given the javadoc in `canOptimizeOutLateralView` method.

Reply:
Code has been changed so canOptimizeLateralView is not needed anymore